api/src/test/java/org/apache/iceberg/expressions/TestStringLiteralConversions.java

```diff
@@ -132,6 +132,38 @@ public void testStringToTimestampLiteral() {
         .isEqualTo(avroValue);
   }
 
+  @Test
+  public void testStringToTimestampLiteralWithMicrosecondPrecisionFromNanoseconds() {
```

Comment:
What is this testing? It uses micros.

Reply:
It tests parsing a nanosecond-precision timestamp to `Literal<CharSequence>` and then converting that to `Literal<Long>` with microsecond precision.